polkadot/node/network/protocol/src/request_response/v2.rs

```diff
@@ -78,3 +79,60 @@ impl IsRequest for CollationFetchingRequest {
 	type Response = CollationFetchingResponse;
 	const PROTOCOL: Protocol = Protocol::CollationFetchingV2;
 }
+
+/// Request an availability chunk.
+#[derive(Debug, Copy, Clone, Encode, Decode)]
+pub struct ChunkFetchingRequest {
+	/// Hash of candidate we want a chunk for.
+	pub candidate_hash: CandidateHash,
+	/// The validator index we are requesting from. This may not be identical to the index of the
```

Comment:
IIUC, in practice only backers can answer for all validator indices, all other nodes will only answer if they see their own validator index in the request. This seems to be like this for compatibility reasons until we make the switch to the new mapping. Did we also explore the options of having the ability to request multiple chunks at the same time ? 

If that's the case it would be a good time to introduce it even if we don't use it now. Otherwise this v2 request is same as v1 just with different semantics.

Reply:
> IIUC, in practice only backers can answer for all validator indices, all other nodes will only answer if they see their own validator index in the request. This seems to be like this for compatibility reasons until we make the switch to the new mapping.

correct

> Did we also explore the options of having the ability to request multiple chunks at the same time ?

> If that's the case it would be a good time to introduce it even if we don't use it now. Otherwise this v2 request is same as v1 just with different semantics.

I wouldn't do that now, because the network protocol was part of the RFC that was voted on and approved. Besides, we don't know yet how that would look like to enable future optimisations.
It could even be a separate network protocol, maybe we'd do some kind of eager sharing of chunks (using a notification instead of a request). Or maybe we'd want to represent the indices as bitfields. Many possibilities here IMO